Add option to stack to use merge rather than rebase

yashNaN opened the pull request:
Stacky should work just fine with either rebasing or git merging. 

This makes a change that adds a `[GIT]` section to the stacky config and 2 settings `use_merge` and `use_force_merge` to change to merge commits and disallowing force pushes. If we disallow force pushes and want to use merging we should also disallowing amending commits, which this pr does.

Sample `~/.stackyconfig` with these changes set 

```
  [UI]
  change_to_main = True
  
  [GIT]
  use_merge = True
  use_force_push = False
``` 

Tested by creating a stack of 3 branches using normal stacky commands

mpatou commented:
Looks good to me, I wished you could add some test to cover, we don't have yet too many  tests but it would be good.
I would like to get the opinion of @tudor too on allowing merge.
I don't like merge we do protect against that, I also understand it's off by default but ...

yashNaN commented:
Yeah I have the opposite issue, my team disallows rebasing and only allows merging. So I would need this to make stacky work with that. 

I kind of buy the argument. `git log --first-parent` will let you see the commits like you expect. And in github you can avoid force-pushing so that when reviewing you can check only the changes since the last time you've reviewed.

mpatou commented on src/stacky/stacky.py:
```diff
@@ -19,6 +19,7 @@
 # onwards). After updating a parent branch P, given a child branch C,
 # we rebase everything from C's PC until C's tip onto P.
 #
+#
```
nit: not sure that this is really needed I guess more comments line won't hurt
